src/index.js

```diff
@@ -32,8 +86,10 @@ module.exports = (
       return values;
     }
 
+    const data = applySerializers(values, serializers);
+
     const blacklistedKeys = new Set();
-    const obj = JSON.parse(stringify(values));
+    const obj = JSON.parse(stringify(data));
```

Comment:
The method `applySerializers` already does a `cloneDeep`, so I'm guessing this is no longer needed.

Reply:
1. Circular references can be checked during traverse.
2. Setters are undefined, any reason not to just delete properties that are undefined?
3. Classes don't seem to affect traversal, do you have any practical case?

Reply:
Our goal was to implement this feature without having to change the input received by traverse and reduce the chances of breaking something. So, I looked deeper into the cases above and classes are the most critical. There goes an example:

```
class Foobar {
        get bar() {
          return 'biz';
        }

        toJSON() {
          return {
            bar: this.bar
          };
        }
}

const anonymize = anonymizer({ whitelist: ['foo.bar'] });
const foo = new Foobar();
anonymize({ foo })
```

With the current approach, the output would be `{ foo: { bar: 'biz' } }`. If we remove the second copy (`JSON.parse(stringify(data))`), we will lose the ability to return this output. The traverse doesn't navigate over the getters of a class, so anonymize won't be able to whitelist or blacklist paths inside classes. The output would be `{ foo: '--REDACTED--' }`.